### PR TITLE
Add basic CI

### PR DESCRIPTION
**Pull request by sergejparity**
PR adds simple CI to perform `cargo test --workspace --features runtime-benchmarks`

### CODE REVIEW

**Comment by ggwpez** [comment]
So I will close this? https://github.com/polkadot-fellows/runtimes/pull/4

**Comment by sergejparity** [comment]
@ggwpez yes, we don't need it

**Comment by ggwpez** [comment]
@sergejparity I added some ideas here https://forum.parity.io/t/repo-reorg/1455/38  
Very basic though.

Running on merge main and PRs is good :+1:

**Review by bkchr — APPROVED** [review]
Looks good!

**Review by ggwpez — COMMENTED** [review]
Will try to make a sync-up to latest master and see if it goes green on another MR.

File: .github/workflows/clippy.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          override: true
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Fetch cache
+        uses: Swatinem/rust-cache@6fd3edff6979b79f87531400ad694fb7f2c84b1f # v2.2.1
+
+      - name: Clippy
+        run:  cargo clippy
```
[the remark this answers is not in the document]
Comment (rcny) [reply .github/workflows/clippy.yml]: Made it the same as in Polkadot CI.

**Comment by rcny** [comment]
`clippy` [still fails](https://github.com/polkadot-fellows/runtimes/actions/runs/4557384571/jobs/8038941879?pr=5) even when invoked as `cargo +nightly clippy --all-targets` with `SKIP_WASM_BUILD=1`

**Comment by rcny** [comment]
`Test all features` [fails](https://github.com/polkadot-fellows/runtimes/actions/runs/4557384569/jobs/8038941849?pr=5) too. AFAIU this is not CI-related.

**Comment by ggwpez** [comment]
Yep Polkadot is actually broken :smile:  
Will fix tomorrow.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          target: wasm32-unknown-unknown
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Fetch cache
+        uses: Swatinem/rust-cache@6fd3edff6979b79f87531400ad694fb7f2c84b1f # v2.2.1
+
+      - name: Test all features
+        run: cargo test --workspace --all-features
```
Comment (ggwpez) [inline .github/workflows/test.yml:44]: Okay the `all-features` was maybe not as smart as first thought. Can we try the *exact* same as in Polkadot? 
  Will also need `RUSTFLAGS: "-Cdebug-assertions=y -Dwarnings"`. 
```suggestion
        run: cargo test --workspace --profile testnet --locked --features=runtime-benchmarks,runtime-metrics,try-runtime
```
 For whatever reason it compiles fine with `try-runtime` enabled but not with `--all-features` :see_no_evil:

Comment (ggwpez) [reply .github/workflows/test.yml:44]: This worked locally for me.

Comment (rcny) [reply .github/workflows/test.yml]: Done, applied.

**Comment by rcny** [comment]
What about Clippy?

**Comment by ggwpez** [comment]
> What about Clippy?

Yea checking that now.

**Review by ggwpez — APPROVED** [review]
With the clippy it looks like version incompatibilities.  
So we can probably merge it red and then try to make it green on the next sync-up with master or on the final import step.

**Comment by rcny** [comment]
Okay, merging then!

**Pull request merged**